arch/arm/core/aarch32/cortex_a_r/__aeabi_read_tp.S

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <toolchain.h>
+
+_ASM_FILE_PROLOGUE
+
+GTEXT(__aeabi_read_tp)
+
+GDATA(z_arm_tls_ptr)
+
+SECTION_FUNC(text, __aeabi_read_tp)
```

Comment:
Is there any chance that you could instead make this as a cmsis function, probably putting it in include/arch/arm/aarch32/cortex_a_r/cmsis.h until it could be moved to the true arm cmsis module?  I assume you probably copied this structure from Cortex-M, but since this uses a mrc, it makes sense to abstract this access.

Reply:
The other side of this operation, storing the TLS value into the register, is done in swap_helper.S without any abstraction, so I'm not sure it makes sense to create an abstraction here?

I also don't think we should use this function anywhere; it's a part of EABI, but should only be used in code compiled with -mtp=soft. However, as Zephyr is the code base storing the TLS base address, it makes sense for Zephyr to offer the standard ABI that also fetches it, so I went ahead and added it.